### PR TITLE
Address policheck bugs

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**

### CODE REVIEW

File: DESIGN-NOTES.md [hunk 1]
```diff
@@ -38,11 +38,11 @@ Instrumentation Engine allows multiple profilers co-exist in the same process. I
...
 
-Sticking with the Intellitrace terminology, these instrumentation components called "Instrumentation Methods" which will essentially be free threaded com objects loaded by direct call to the class factory using the com server exports (DllGetClassObject etc…). This has the advantage of requiring a simple vtable dispatch for each cross component call keeping overhead to a minimum.
+Sticking with the IntelliTrace terminology, these instrumentation components called "Instrumentation Methods" which will essentially be free threaded com objects loaded by direct call to the class factory using the com server exports (DllGetClassObject etc…). This has the advantage of requiring a simple vtable dispatch for each cross component call keeping overhead to a minimum.
```
Comment (delmyers) [inline DESIGN-NOTES.md]: >IntelliTrace [](start = 18, length = 12)

Sorry, I never looked at the initial check-in. Why is it that we mention IntelliTrace in this so much? Seems like a historical detail that isn't much use to people who are trying to write instrumentation engines.

File: DESIGN-NOTES.md [hunk 2]
```diff
@@ -38,11 +38,11 @@ Instrumentation Engine allows multiple profilers co-exist in the same process. I
...
+In order to allow for xcopy deploy without admin rights, the registration of the instrumentation components is done via an object configuration model using xml files. This is similar to how IntelliTrace manages instrumentation methods today with one key difference: the instrumentation objects will likely be packaged in their own dlls. Therefore, for MSI based installations, a safe admin writable folder will contain the dlls, and the configuration file will describe the objects that are contained within that dll. For xcopy deployment, the sub folder containing the instrumentation methods will be a sub-folder of the xcopy payload and the xml config will be contained in the same folder as the monitoring agent.
```
Comment (delmyers) [inline DESIGN-NOTES.md]: >ntelliTrace manages instrumentation meth [](start = 191, length = 40)

This is an internal implementation detail of IntelliTrace, and it doesn't really map to what happens in CLRInstrumentationEngine. I would take out references to IntelliTrace.

Comment (delmyers) [inline DESIGN-NOTES.md]: > monitoring agent [](start = 698, length = 17)

monitoring agent? What monitoring agent? This is not part of CLRInstrumentationEngine.

The design of CLRInstrumentationEngine does not require that the Instrumentation Methods be in a secure folder any more than the CLR currently require s that the ICorProfiler implementation needs to be in a secure folder. All that is required is that the folder/files need to be readable by the instrumented process, and that (on Windows) the instrumentation method module is signed.

File: DESIGN-NOTES.md [hunk 3]
```diff
@@ -110,15 +110,15 @@ Many profiler consumers will need callbacks when interesting events happen in th
...
 4. OnClassLoad
-    * Intellitrace uses this callback which occurs every time a new class loads to update the type system. This callback MUST be optional as it incurs high overhead
+    * IntelliTrace uses this callback which occurs every time a new class loads to update the type system. This callback MUST be optional as it incurs high overhead
     * Parameters:
         * Method Info
         * Class Info?
         * IMetadataImport
 5. Function Enter / Function Exit
-    * These callbacks are issued every time a function is called and every time a function returns. Intellitrace and Production BPs don’t use these, but Tofino does. Note that these also incur large runtime overhead and use a naked calling convention making it difficult to multiplex
+    * These callbacks are issued every time a function is called and every time a function returns. IntelliTrace and Production BPs don’t use these, but Tofino does. Note that these also incur large runtime overhead and use a naked calling convention making it difficult to multiplex
```
Comment (delmyers) [inline DESIGN-NOTES.md]: >Production BPs [](start = 117, length = 14)

Also remove references to Production BPs. These are implementation details of that product, not CLRInstrumentationEngine.

Comment (delmyers) [inline DESIGN-NOTES.md]: >Tofino [](start = 153, length = 6)

Defunct, internal MS project name.

File: DESIGN-NOTES.md [hunk 4]
```diff
@@ -110,15 +110,15 @@ Many profiler consumers will need callbacks when interesting events happen in th
...
 6. Inline tracking
```
Comment (delmyers) [inline DESIGN-NOTES.md]: > Inline tracking [](start = 2, length = 16)

Inlining also limits the things that the profiler can do. There are limitations with unloadable code, for example. We should probably document the limitations of inlining.

File: DESIGN-NOTES.md [hunk 5]
```diff
@@ -110,15 +110,15 @@ Many profiler consumers will need callbacks when interesting events happen in th
...
-    * This allows components to track where a method has been inlined. Intellitrace turns inlining off globally today. Tofino allows inlining to occur.
+    * This allows components to track where a method has been inlined. IntelliTrace turns inlining off globally today. Tofino allows inlining to occur.
     * Inlining greatly complicates the way the profiler works. However, the increase in performance is likely worth building something into the high level structures to make it easy to go from a metadata token to its inline sites to allow consumers to easily rejit all of the parent function to undo inlining.
 
 Instrumentation Methods implements the ICorProfilerCallback interfaces (2, 3, 4, 5) to receive notifications of these events from the profiler manager. The profiler manager will expose the higher level api to allow the instrumentation methods to co-exist and make writing them easier. For instance, when receiving a jit callback, instrumentation methods can query the ProfilerManager to map MethodId to an instance of IMethodInfo and use that to get the instruction graph, examine metadata etc…
```
Comment (delmyers) [inline DESIGN-NOTES.md]: >2, 3, 4, 5 [](start = 72, length = 10)

It implements more than this now :-)

File: DESIGN-NOTES.md [hunk 6]
```diff
@@ -132,7 +132,7 @@ Instrumentation Methods should call `IProfilerManager::SetEventMask` to specify
 
 In the current ICorProfiler design, there can only be one single function enter, function exit, and function tail call callback function. This is no longer the case with this new model. Instrumentation methods that wish to receive these callbacks can do so enabling the event mask on the Profiler Manager and then setting the callbacks passing the clsid of the instrumentation method. The ProfilerManager will multiplex the callbacks to each of the interesting calls. Similarly, each instrumentation method can register for the FunctionIDMapper  to allow them to disable the callback for a particular method. If no instrumentation method one states that the callback should occur, it is disabled. However, if at least one leaves it enabled, the callback will still occur but will filter out instrumentation methods that do not wish to receive it.
```
Comment (delmyers) [inline DESIGN-NOTES.md]: >there can only be one single function enter, function exit, and function tail call callback [](start = 36, length = 91)

CLRInstrumentationEngine doesn't currently support ELT.

**Comment by delmyers** [comment]
I get what you are trying to do with this PR, but I think that this whole document needs a rewrite. It is old, written from the perspective of "this is how we ought to design the API" rather than, "This is what the API does".

**Comment by WilliamXieMSFT** [comment]
Appreciate the comments you've provided, Del!

---
In reply to: [589252444](https://github.com/microsoft/CLRInstrumentationEngine/pull/206#issuecomment-589252444) [](ancestors = 589252444)

**Comment by WilliamXieMSFT** [comment]
#207

**Pull request merged**